src/frontend/src/optimizer/plan_node/logical_hop_window.rs

```diff
@@ -317,8 +322,35 @@ impl ColPrunable for LogicalHopWindow {
 
 impl PredicatePushdown for LogicalHopWindow {
     fn predicate_pushdown(&self, predicate: Condition) -> PlanRef {
-        // TODO: hop's predicate pushdown https://github.com/risingwavelabs/risingwave/issues/6606
-        gen_filter_and_pushdown(self, predicate, Condition::true_cond())
+        // Do not push down `window_start`, `window_end` for now, they are columns produced by
+        // HopWindow.
+        let mut window_columns = FixedBitSet::with_capacity(self.output_indices().len());
```

Comment:
Should this be `self.schema().len()` (Or `self.input().schema().len()+2`)
```suggestion
        let mut window_columns = FixedBitSet::with_capacity(self.schema().len());
```

Is it possible that `output_indices` becomes smaller than length of schema? (TBH I forgot what's its usage 😇

Reply:
> Should this be self.schema().len() (Or self.input().schema().len()+2)

Agreed, I think `self.schema().len()` seems more consistent.

> Is it possible that output_indices becomes smaller than length of schema? (TBH I forgot what's its usage 😇

From tracing in: https://github.com/risingwavelabs/risingwave/pull/6910
`output_indices` maps input schema (`self.input().schema()`) fields to output schema (`self.schema()`):
```
self.output_indices(): [0, 1, 2, 4, 5]

self.schema(): 
Schema { fields: [auction.id:Int64, auction.item_name:Varchar, 
auction.date_time:Timestamp, auction.seller:Int64, auction.extra:Varchar] }

self.input().schema(): 
Schema { fields: [auction.id:Int64, auction.item_name:Varchar, 
auction.date_time:Timestamp, auction.expires:Timestamp, auction.seller:Int64, auction.extra:Varchar] }
```

cc @st1page is this correct?

Reply:
actually, it is maps `self.input().schema() + [window_start] + [window_end]` to the schema (`self.schema()`)

Reply:
Oh yeah missed that from pruning steps, thanks!

Full trace (https://github.com/risingwavelabs/risingwave/pull/6910) with window_start and end included in output_indices:
```
self.output_indices(): [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]
self.schema(): Schema { fields: [auction.id:Int64, auction.item_name:Varchar, auction.description:Varchar, auction.initial_bid:Int64, auction.reserve:Int64, auction.date_time:Timestamp, auction.expires:Timestamp, auction.seller:Int64, auction.category:Int64, auction.extra:Varchar, window_start:Timestamp, window_end:Timestamp] }
self.input().schema(): Schema { fields: [auction.id:Int64, auction.item_name:Varchar, auction.description:Varchar, auction.initial_bid:Int64, auction.reserve:Int64, auction.date_time:Timestamp, auction.expires:Timestamp, auction.seller:Int64, auction.category:Int64, auction.extra:Varchar] }

self.output_indices(): [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]
self.schema(): Schema { fields: [auction.id:Int64, auction.item_name:Varchar, auction.description:Varchar, auction.initial_bid:Int64, auction.reserve:Int64, auction.date_time:Timestamp, auction.expires:Timestamp, auction.seller:Int64, auction.category:Int64, auction.extra:Varchar, window_start:Timestamp, window_end:Timestamp] }
self.input().schema(): Schema { fields: [auction.id:Int64, auction.item_name:Varchar, auction.description:Varchar, auction.initial_bid:Int64, auction.reserve:Int64, auction.date_time:Timestamp, auction.expires:Timestamp, auction.seller:Int64, auction.category:Int64, auction.extra:Varchar] }

self.output_indices(): [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]
self.schema(): Schema { fields: [auction.id:Int64, auction.item_name:Varchar, auction.description:Varchar, auction.initial_bid:Int64, auction.reserve:Int64, auction.date_time:Timestamp, auction.expires:Timestamp, auction.seller:Int64, auction.category:Int64, auction.extra:Varchar, window_start:Timestamp, window_end:Timestamp] }
self.input().schema(): Schema { fields: [auction.id:Int64, auction.item_name:Varchar, auction.description:Varchar, auction.initial_bid:Int64, auction.reserve:Int64, auction.date_time:Timestamp, auction.expires:Timestamp, auction.seller:Int64, auction.category:Int64, auction.extra:Varchar] }

self.output_indices(): [0, 1, 2, 4, 5]
self.schema(): Schema { fields: [auction.id:Int64, auction.item_name:Varchar, auction.date_time:Timestamp, auction.seller:Int64, auction.extra:Varchar] }
self.input().schema(): Schema { fields: [auction.id:Int64, auction.item_name:Varchar, auction.date_time:Timestamp, auction.expires:Timestamp, auction.seller:Int64, auction.extra:Varchar] }
```